go/vt/vttablet/tabletmanager/vreplication/replicator_plan.go

```diff
@@ -646,6 +654,24 @@ func appendFromRow(pq *sqlparser.ParsedQuery, buf *bytes2.Buffer, fields []*quer
 				buf.WriteString(sqltypes.NullStr)
 			} else {
 				vv := sqltypes.MakeTrusted(typ, row.Values[col.offset:col.offset+col.length])
```

Comment:
Does this also allocate and later on too? Is it worth avoiding creating this if we overwrite it later?

Reply:
Done. No double allocation. Also, converged the two codepaths that do `charset.Convert()` into a single `convertStringCharset()` function.